build.go

```diff
@@ -97,6 +102,16 @@ func (vs *version) addFiles(partitions map[int]bool) error {
 }
 
 func (vs *version) addFile(file string, partitions map[int]bool) error {
+	if vs.stats != nil {
+		start := time.Now()
+
+		defer func() {
+			duration := time.Since(start)
+			vs.stats.Timing("s3.download_duration", duration, []string{vs.db.name}, 1)
+			vs.stats.Decr("s3.queue_depth", []string{vs.db.name}, 1)
```

Comment:
Similarly, here, don't emit a decrement at all; just emit the full size of the value as a gauge when it changes/periodically/whatever makes sense. Making increments and decrements line up over time in DD dashboards is near impossible, and they don't come across as expected usually